### DISCUSSION TITLE
left sidebar

### DISCUSSION THREAD

**Discussion by VGOVICH** [bootscore/bootscore#815] [Q&A]
I brought up the right menu in the sidebar via the menu widget. Tell me how to make it on the left side? There is no clear information anywhere.

**Comment by VGOVICH** [comment]
`I also tried to create a vertical left menu. The settings appeared, but nothing was selected on the site.
functions.php

```
function register_my_menus() {
    register_nav_menus(
        array(
            'primary-menu' => __( 'Primary Menu' ),
        )
    );
}
add_action( 'after_setup_theme', 'register_my_menus' );
```


left-sidebar.php

```
<aside id="left-sidebar" class="col-md-3">
    <?php if ( is_active_sidebar( 'left-sidebar' ) ) : ?>
        <?php dynamic_sidebar( 'left-sidebar' ); ?>
    <?php else : ?>
      
        <section class="widget">
            <h2 class="widget-title"><?php esc_html_e( 'Widget Area', 'bootscore' ); ?></h2>
            <p><?php esc_html_e( 'You can add widgets here.', 'bootscore' ); ?></p>
        </section>
    <?php endif; ?>

    
    <nav class="sidebar-menu">
        <?php
            wp_nav_menu( array(
                'theme_location' => 'primary-menu',
                'menu_id'        => 'primary-menu',
                'menu_class'     => 'menu',
            ) );
        ?>
    </nav>
</aside>
```

**Reply by crftwrk** [reply]
### Sidebar

For posts and pages you can choose the [sidebar-left](https://bootscore.me/theme/page-sidebar-left/) template in the backend edit screen.

If you want to have the sidebar in general on the left side, it depends if you are on v5 or v6. If you are still on v5, you can override [templates](https://bootscore.me/documentation/templates/) in your child and change the col ordering.

If you are on v6, simply use [filters](https://bootscore.me/documentation/filters/#main-col-sidebar-and-breakpoints) in child's `functions.php` to change the sidebar to the left in all templates without overrides:

```php
/**
 * Change sidebar col
 */
function change_sidebar_col_size($col_size) {
  return "col-lg-3 order-first";
}
add_filter('bootscore/class/sidebar/col', 'change_sidebar_col_size');


/**
 * Change sidebar responsive offcanvas
 */
function change_sidebar_offcanvas($col_size) {
  return "offcanvas-lg offcanvas-start";
}
add_filter('bootscore/class/sidebar/offcanvas', 'change_sidebar_offcanvas');
```

### Menu

Register a new menu in child's `functions.php`:

```php
/**
 * Register additional menu
 */
register_nav_menu('primary-menu', 'Primary Menu');
```

Now you can use a legacy navigation widget (not recommended) to show a simple `ul` in sidebar widget. If you want to use a "real" menu then you have to register the new menu in template file as well.

Override the `sidebar.php` in your child and add the new menu:

```php
<?php
  wp_nav_menu(array(
    'theme_location' => 'primary-menu',
    'container' => false,
    'menu_class' => '',
    'fallback_cb' => '__return_false',
    'items_wrap' => '<ul id="%1$s" class="nav flex-column">%3$s</ul>',
    'depth' => 2,
    'walker' => new bootstrap_5_wp_nav_menu_walker()
  ));
?>
```

Tested and works fine. Solved?

**Comment by VGOVICH** [comment]
I need to display a menu in the left column using a widget and your code. But, unfortunately, the default menu cannot be displayed without vertical widgets.

![Screenshot_13](https://github.com/user-attachments/assets/cb0c136c-df44-4248-a88f-efea371b664c)

I set the settings in the Wordpress console to Primary Menu. 
 * @package Bootscore
 * @version 6.0.0
 */

**Reply by crftwrk** [reply]
No, as I explained in my previous comment, you have to register the `nav_walker` in the `sidebar.php` template es well if you want to do it that way. That's because the backend must know where the menu is. This is not widget area.

If you want to use a widget for a menu, then you must use a legacy widget, because Gutenberg does not support menus in widgets. In that case, you do not need to register a new menu in `functions.php`. Just create a new menu in the backend and leave menu position checkbox unchecked.

1. Add a new menu in the backend and leave position checkbox unchecked 
2. Go to widget area
3. Press + to add a new block
4. Search for legacy
5. Choose and select Navigation Menu
6. Select Menu you want to show in the dropdown


| First Header  | Second Header |
| ------------- | ------------- |
| ![legacy-1](https://github.com/user-attachments/assets/6fa58c94-a17b-46a9-aa56-cf78498fa17a)  | ![legacy-2](https://github.com/user-attachments/assets/a889593a-9ebc-4361-a919-ae970c782c89)  |
| ![legacy-3](https://github.com/user-attachments/assets/2dc98550-da26-494c-814f-fe5687c1c646)  |   |

**Comment by VGOVICH** [comment]
Thank you, the issue has been resolved. The toric can be closed.

чт, 25 июл. 2024 г., 21:02 Basti ***@***.***>:

**Reply by crftwrk** [reply]
You‘re welcome 👍 Please mark the answer as solved.

**Comment by VGOVICH** [comment]
Thanks, I will!

пт, 26 июл. 2024 г. в 00:02, Basti ***@***.***>:
